specs/custom-iresourcemanager-spec.md

```diff
@@ -0,0 +1,109 @@
+Providing a custom MRT Core `IResourceManager` for WinUI 3 to use
+===
+
+# Background
+
+The WinUI 3 framework instantiates an MRT Core
+[`ResourceManager`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.applicationmodel.resources.resourcemanager?view=windows-app-sdk-1.2)
+in order to resolve resource URIs. While this default `ResourceManager` is generally sufficient, 
+there are some scenarios in which an app needs non-standard behavior in order to resolve a
+particular resource URI. In order to address this requirement, this document describes a new
+API that would allow an app to provide its own custom implementation of the
+[`IResourceManager`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.applicationmodel.resources.iresourcemanager?view=windows-app-sdk-1.2)
+for WinUI 3 to use in lieu of its standard `ResourceManager`.
+
+# API Pages
+
+_(Each of the following L2 sections correspond to a page that will be on docs.microsoft.com)_
+
+## Application.ResourceManagerRequested Event
+
+Raised during startup of a new WinUI thread to give the app a chance to provide its own
+[`IResourceManager`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.applicationmodel.resources.iresourcemanager?view=windows-app-sdk-1.2)
+implementation to be used by the framework for resolving resource URIs.
+
+_Spec note: is there a version of the 'ResourceDictionary and XAML resource references' article that 
+links to the Windows App SDK documentation for relevant APIs?_
+
+```c#
+public event TypedEventHandler<Application, ResourceManagerRequestedEventArgs> 
+ResourceManagerRequested
+
+```
+
+### Remarks
+
+It is strongly recommended that apps which need to listen for this event register their handler in their `App` 
+class's constructor so that it is available during initial app launch. This event is raised once per WinUI
+thread during initialization. Because the framework assumes that each `IResourceManager` instance returned
+through the event is independent, it is your responsibility to ensure that the implementation is thread-safe
+if you wish to share the same instance between multiple threads.
```

Comment:
"Thread" was the term we settled on in place of "core" since the latter is not defined/explained well (at all?) in external documents. I don't know what correlation there is between that and COM apartments but as I understand it apartments live at a much lower abstraction layer and so wouldn't be expected to come up in this context.

Reply:
> the framework assumes that each `IResourceManager` instance returned through the event is independent

"Independent" is an unusual word to use in this context, and it took me some time to figure out what it meant. Maybe less confusing would be to remove the discussion of "independent" and just say

> If you use the same `IResourceManager` for multiple threads, the `IResourceManager` must be thread-safe.